apple/REAModule.mm

```diff
@@ -295,7 +279,7 @@ - (void)sendEventWithName:(NSString *)eventName body:(id)body
 
 RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(installTurboModule : (nonnull NSString *)valueUnpackerCode)
 {
-  if (!self.bridge) {
+  if (_isBridgeless) {
```

Comment:
Why do you need a separate implementation for bridgeless here?

The implementation for non-bridgeless looks good to me, just don't use `[self.bridge respondsToSelector:@selector(runtime)]` check – it no longer returns `YES` even if the runtime is there. The reason is that the bridge is now an instance of `RCTBridgeProxy` that inherits from `NSProxy` which probably has a custom implementation for `respondsToSelector`, different than `NSObject`.
We know that `bridge.runtime` has been available (as long as you have a category interface) for many RN versions back, so it seems to be safe to call it.

Two more nitpicks regarding the implementation for bridgeless:
- Using `[RCTBridge currentBridge]` is very risky and I guess pretty easy to break in an app with multiple bridges (or now React instances).
- `bridge.jsCallInvoker` was initially not available in the bridge proxy, but it is now. I think there is no need to use the runtime executor in its current shape (call invoker, runtime executor and runtime scheduler might be unified in the next RN versions).

Reply:
> bridge.jsCallInvoker was initially not available in the bridge proxy, but it is now. I think there is no need to use the runtime executor in its current shape (call invoker, runtime executor and runtime scheduler might be unified in the next RN versions).

Are you telling me that all our work on refactoring Reanimated initialization pipeline to be able to dynamically utilize either `jsCallInvoker` or `runtimeExecutor` was mostly pointless? 💀

Reply:
Yes xD

Maybe not entirely pointless, because the runtime executor might become the recommended way when the RN team figures out the final API for libraries. Right now we have two ways and both are lacking of features (e.g. synchronous executions), so I would recommend to use the call invoker and don't complicate the code too much until then.